src/Lazy/Tree.elm

```diff
@@ -474,10 +473,38 @@ fromList_ parent isParent list =
     LL.llist (List.map (constructTree isParent list) << List.filter (isParent parent)) list
 
 
-{-| TODO better name if we decide to include it
+{-| Construct `Forest` from a list of items that can be **uniquely identified** by comparable value.
+
+This function can yield much better performance than more general [`fromList`](#fromList) alternative.
+Be aware that this function pushes **more work on initial construction of the Tree** at the benefit of
+doing more optimal operations during evaluation of the Tree at later time.
+
+    import Lazy.LList as LL
+
+    [ { id = 1, parent = Nothing }
+    , { id = 2, parent = Nothing }
+    , { id = 3, parent = Just 1 }
+    ]
+        |> fromKeyedList .id .parent
+        |> LL.map (.id << item)
+        |> LL.toList
+    --> [ 1, 2 ]
+
+    [ { id = 1, parent = Nothing }
+    , { id = 2, parent = Nothing }
+    , { id = 3, parent = Just 1 }
+    , { id = 4, parent = Just 1 }
+    , { id = 5, parent = Just 2 }
+    ]
+        |> fromList (\p i -> Maybe.map .id p == i.parent)
+        |> LL.andThen descendants
+        |> LL.map (.id << item)
+        |> LL.toList
+    --> [ 3, 4, 5 ]
+
 -}
-fromListWithComparableIds : (a -> comparable) -> (a -> Maybe comparable) -> List a -> Forest a
-fromListWithComparableIds getNodeId getParentId list =
+fromKeyedList : (a -> comparable) -> (a -> Maybe comparable) -> List a -> Forest a
```

Comment:
@jhrcek one thing to think about is if it wouldn't be possible to define this function in a way it's possible for single item to belong to multiple parents.